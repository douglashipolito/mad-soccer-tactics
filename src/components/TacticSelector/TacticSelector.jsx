import React from "react";
import { Consumer } from "../Tactics/TacticsContext";

import "./TacticSelector.scss";

class TacticSelector extends React.Component {
  render() {
    if (!this.props.team) {
      return null;
    }

    return (
      <Consumer>
        {context => (
          <div className="tactice-selector-component">
            <form>
              <label htmlFor={`tactic-selector-${this.props.team.id}`}>
                <span>Tactic</span>
                {/* eslint-disable-next-line jsx-a11y/no-onchange */}
                <select
                  onChange={event =>
                    context.handleTacticChange(event, this.props.team.id)
                  }
                >
                  {this.props.tactics.map(tactic => (
                    <option key={tactic.id} value={tactic.id}>
                      {tactic.name}
                    </option>
                  ))}
                </select>
              </label>
            </form>
          </div>
        )}
      </Consumer>
    );
  }
}

export default TacticSelector;
