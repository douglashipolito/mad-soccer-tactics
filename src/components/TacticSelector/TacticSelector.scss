.tactice-selector-component {
  span {
    margin-right: 10px;
  }
}
