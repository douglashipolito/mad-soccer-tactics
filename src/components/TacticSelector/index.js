import TacticSelector from "./TacticSelector.jsx";

export default TacticSelector;
