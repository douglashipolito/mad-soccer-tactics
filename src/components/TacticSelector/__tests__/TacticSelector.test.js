import React from "react";
import { create } from "react-test-renderer";
import TacticSelector from "../";

test("Tactic Selector Renders", () => {
  const string = create(<TacticSelector />);
  expect(string).toMatchSnapshot();
});
