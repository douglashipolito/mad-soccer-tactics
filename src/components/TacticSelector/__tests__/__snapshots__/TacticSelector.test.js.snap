// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tactic Selector Renders 1`] = `null`;
